docker-compose.yml

```diff
@@ -0,0 +1,84 @@
+# Docker Compose file for the services needed by Ocim, for easy local development setup
+
+version: "3.6"
+
+services:
+
+  mongo:
+    # This is similar to what the EdX devstack uses
+    # We use WiredTiger in all environments. In development environments we use small files
+    # to conserve disk space, and disable the journal for a minor performance gain.
+    # See https://docs.mongodb.com/v3.0/reference/program/mongod/#options for complete details.
+    command: mongod --smallfiles --nojournal --storageEngine wiredTiger
+    container_name: opencraft.mongo
+    hostname: mongo
+    image: mongo:4.0.22
```

Comment:
I think these are the ones OCIM manages and assign OpenEdX instances to and it's part of the prior work done by @janimo.

I've used these instances for running tests locally but most other people probably use other instances running on OVH, so they're not mandatory.
If you agree, I'll leave this as is *for now* and create an issue for only starting them optionally (otherwise I'm afraid this will slip and kill my onboarding budget).

Reply:
@kewne If you can get it to a working state, then I'd suggest creating a follow-up ticket and placing it under the OCIM account instead, pinging @lgp171188 so it can be prioritized.

Containerization of OCIM's varying components and getting off of Vagrant is something that needs to be done anyway. The M1 is just the impetus for it now. So after you get it up and running in an acceptable fashion, a follow-up in-budget ticket that handles remaining polish would be OK.

Reply:
@Kelketek @bradenmacdonald ok, let me know if there's anything else you'd like to see fixed right now; otherwise I'll just squash and rebase this unto master so it can be merged 🙂

Reply:
+1, I was going to say, this doesn't need to come out of your onboarding budget, this is work we've wanted for a long time anyways.

It's fine to leave these in for now though.

I guess what I don't understand is: if you want to test Ocim, you need a Mongo/MySQL/Rabbit MQ server _and_ you need to be able to deploy VMs that run Open edX. If you are running everything in the OVH cloud, that makes sense, but if you are running Mongo/MySQL/RabbitMQ locally in docker, how can they even be used, unless you also have some way of deploying Open edX VMs onto your localhost, or allowing VMs in the OVH cloud to access your local Mongo/MySQL/RabbitMQ in your laptop's docker environment?

Reply:
@bradenmacdonald Provisioning servers locally hasn't ever worked well from what I know-- it's completely unsupported. If someone added these, I could think of a couple of reasons why:

1. Maybe they didn't understand what the MySQL servers were for yet when first getting the service to run on Docker
2. They did understand, but their goal wasn't to enable provisioning as much as to make sure that the code didn't crash when assuming these services would be present in passing.

I think there's some pretty good argument for 2 because usually one needs to 'provision' an instance just enough to get it to fail so that there's the relevant local DB entries all set up consistently aside from the fact the provisioning itself fails. That would probably involve creating the DB and setting the resulting created credentials in the instance settings.

Hard to say, though.

Reply:
In my particular case, [the story I'm working on](https://tasks.opencraft.com/browse/BB-3729) only required me to create some unit tests which don't provision any VMs...
You totally have a point though that Postgres is the only real OCIM dependency and the rest are transitive dependencies of OpenEdX, so I've removed them.